internal/install/segment/client.go

```diff
@@ -0,0 +1,121 @@
+package segment
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+const (
+	eventName = "VirtuosoCLIInstall"
+)
+
+type EventType string
+
+var EventTypes = struct {
+	InstallStarted          EventType
+	AccountIDMissing        EventType
+	APIKeyMissing           EventType
+	RegionMissing           EventType
+	UnableToConnect         EventType
+	UnableToFetchLicenseKey EventType
+	AbleToFetchLicenseKey   EventType
+	UnableToOverrideClient  EventType
+}{
+	InstallStarted:          "InstallStarted",
+	AccountIDMissing:        "AccountIDMissing",
+	APIKeyMissing:           "APIKeyMissing",
+	RegionMissing:           "RegionMissing",
+	UnableToConnect:         "UnableToConnect",
+	UnableToFetchLicenseKey: "UnableToFetchLicenseKey",
+	AbleToFetchLicenseKey:   "AbleToFetchLicenseKey",
+	UnableToOverrideClient:  "UnableToOverrideClient",
+}
+
+var (
+	embedded embed.FS
+)
+
+type Segment struct {
+	analytics.Client
+}
+
+func New() *Segment {
+	writeKey, err := getWriteKey()
+	if err != nil {
+		log.Warnf("segment: error reading write key, cannot write to segment %v", err)
+		return nil
+	}
+	if writeKey == "" {
+		log.Warn("segment: write key is empty, cannot write to segment")
+		return nil
+	}
+
+	client := analytics.New(writeKey)
+	log.Info("segment initialized")
+
+	return &Segment{client}
+}
+
+func (client *Segment) Track(accountID int, event Event) {
+
+	if client == nil {
+		return
+	}
+
+	properties := toMap(event)
+
+	properties["category"] = "newrelic-cli"
+	properties["accountId"] = accountID
+
+	err := client.Enqueue(analytics.Track{
+		UserId:     fmt.Sprintf("%d", accountID),
+		Event:      eventName,
```

Comment:
@NRhzhao I think there is a mistake here, it should be the eventName passed in from the method parameter. Also, I don't think that const eventName is needed.